mlx/backend/metal/kernels/reduction/reduce_col.h

```diff
@@ -20,170 +15,128 @@ template <
     const constant size_t& non_col_reductions [[buffer(10)]],
     uint3 gid [[threadgroup_position_in_grid]],
     uint3 gsize [[threadgroups_per_grid]],
-    uint simd_lane_id [[thread_index_in_simdgroup]],
-    uint simd_group_id [[simdgroup_index_in_threadgroup]],
-    uint3 tid [[thread_position_in_grid]],
-    uint3 tsize [[threads_per_grid]]) {
+    uint3 lid [[thread_position_in_threadgroup]],
+    uint3 lsize [[threads_per_threadgroup]]) {
+  constexpr int n_reads = 4;
   Op op;
   looped_elem_to_loc<NDIMS> loop;
   const device T* row;
 
-  // Case 1: Small row small column
-  if (reduction_size * non_col_reductions < 64 && reduction_stride < 32) {
-    U totals[31];
-    for (int i = 0; i < 31; i++) {
-      totals[i] = Op::init;
-    }
-
-    short stride = reduction_stride;
-    short size = reduction_size;
-    short blocks = stride / N_READS;
-    short extra = stride - blocks * N_READS;
+  U totals[n_reads];
+  for (int i = 0; i < n_reads; i++) {
+    totals[i] = Op::init;
+  }
```

Comment:
```suggestion
  U totals[n_reads] = {Op::init};
```

It shouldn't really matter, but aleast on my M2 Ultra system with the test `python benchmarks/python/comparative/bench_mlx.py sum_and_add --axis 0 --size 16x4096 --size 16x4096` it gave a small improvement.

Reply:
Yeah, perhaps we could resort to an `if constexpr (Op::init == 0)` because otherwise this is not the same as before (at least in C++, maybe metal has something I missed).

Basically,

```cpp
int x[3] = {0}; // initialize everything to 0
int y[3] = {1}; // initialize the first to 1 and the rest to 0
```

Reply:
Good point, I missed that, yeah we can stick with what it was before, no need for a `if constexpr (Op::init == 0)`